FLEDGE_extended_PA_reporting.md

```diff
@@ -0,0 +1,335 @@
+
+Our proposed API extension allows a bidder to register a set of histogram contributions associated
+with an arbitrary `event_key` within `generateBid`, `scoreAd`, `reportWin`, and `reportResult`.
+
+### Example 1: Correlating bidding signals with click information.
+
+We consider the scenario where a buyer wants to learn the click through rate of ads when a user has
+been in an interest group for a given time.
+
+The buyer may implement `getImpressionReportBucket()` and `getClickReportBucket()` which map an
+interest group and the time the user has spent in that interest group to a 128-bit integer.
+
+The buyer can then do the following during generateBid (when the above information is available)
+
+```
+function generateBid(interestGroup, auctionSignals, perBuyerSignals, trustedBiddingSignals, browserSignals) {
+ …
+  privateAggregation.reportContributionsForEvent({eventType: “reserved.win”, contributions: {
+      bucket: getImpressionReportBucket()
+      value: 1
+  };
+  privateAggregation.reportContributionsForEvent({eventType: "click", contributions: {
+      bucket: getClickReportBuckets(), // 128-bit integer
+      value: 1
+    });
+```
+
+The above logic will trigger a report if the generated bid wins (see reserved.win). And another one,
+if the user later clicks on the winning ad (this needs to be triggered by the fenced frame itself, see
+reporting for post-auction signals). When the buyer receives an aggregated report they can infer what
+the click-through-rate (CTR) was for users on different “interest group age” buckets.
+
+### Example 2: Getting the average bid gap for an ad. 
+
+A buyer may be interested in understanding how much higher they should have bid in order to win an
+auction. To do this, not only do we need to trigger a report at loss time (see Reporting for bids
+which do not win), but also have the value of the report depend on the outcome of the auction. To
+do this, we introduce a field to the “contributions” object called `signalValue`. This field allows
+the report to depend on post auction information. A `signalValue` object is composed of the following
+values:
+* `baseValue`: The name of the auction result value we want to report. For instance, winningBid. 
+* `offset`: Optional offset to add/subtract to the auction result value
+* `scale`: Optional scale factor by which we want to multiply the output. This is useful for
+controlling the amount of noise added by the aggregation service. Scale is applied after `offset`
+is subtracted.
+
+After the auction happens, the final value of the generated report is scale*(baseValue + offset). The
+following example shows  how to return the gap between an ad bid and the winning bid.
+
+```
+function generateBid(...) {
+  bid = 100
+  privateAggregation.reportContributionsForEvent({
+    eventType: "reserved.loss"
+    contributions: {
+      bucket: 1596 // represents a bucket for interest group x winning bid price
+      value: {
```

Comment:
```suggestion
      signalValue: {
```

Reply:
I think the idea here is that the `value` field is a union of a integer OR a `signalValue`, leaving as-is.